### PR TITLE
Implemented auto loader for Inject

### PR DESCRIPTION
**Pull request by wojciech-kulik**
What do you guys think about this solution? It removes the need of using `Inject.load` or `.enableInjection()` (unless you want live changes) :)

### CODE REVIEW

**Comment by johnno1962** [comment]
Hi, This seems like a good idea and is effectively what the iOSInjectioBundle does on loading itself. I don't think you can encourage people to do without `.enbleInjection()` though as it also erases the type of content Views which is critical for reliable SwiftUI updates. It may appear to work most of the time but add a UI element that alters the concrete type underlying the Opaque `some View` and you'll get a crash.

**Comment by wojciech-kulik** [comment]
Yes, `enableInjection` will be required for SwiftUI if you want live changes. However, for example in UIKit project I don't always need live changes, what's more, I don't always use Inject for UI hot reload, sometimes just for tweaking code and thanks to these changes I won't have to change anything in my code to enable Inject. Before I had to add in `AppDelegate` `_ = Inject.load`.

The same applies to SwiftUI, now if you want to have a hot reload, you need to navigate to a screen that uses `.enableInjection` or enable it explicitly by `_ = Inject.load`. This way you have hot reloaded enabled as soon as app starts :)

**Comment by wojciech-kulik** [comment]
I agree, that's why we should keep the information in README to use always `.enableInjection()` for SwiftUI views. However, this will be a nice addition having a hot reload when the app starts.

File: Sources/Inject/Inject.swift [hunk 1]
```diff
@@ -7,22 +7,19 @@ import SwiftUI
...
 }
 
 /// Public namespace for using Inject API
 public enum Inject {
     public static let observer = injectionObserver
-    public static let load: Void = loadInjectionImplementation
     public static var animation: SwiftUI.Animation?
 }
 
-public extension InjectListener {
```
Comment (krzysztofzablocki) [inline Sources/Inject/Inject.swift:21]: you need to restore this, or release builds will break

File: Sources/Inject/Integrations/SwiftUI.swift [hunk 2]
```diff
@@ -15,7 +13,6 @@ public extension SwiftUI.View {
         onReceive(Inject.observer.objectWillChange, perform: {
             callback(self)
         })
-        .enableInjection()
```
Comment (krzysztofzablocki) [inline Sources/Inject/Integrations/SwiftUI.swift:18]: you still want type erasure here so leave enableInjection?

File: Sources/InjectAutoLoader/InjectAutoLoader.m [hunk 3]
```diff
@@ -0,0 +1,24 @@
+//
```
Comment (krzysztofzablocki) [inline Sources/InjectAutoLoader/InjectAutoLoader.m:1]: can we remove copyright entry? I don't have one even on original lib files 😂

Comment (wojciech-kulik) [reply Sources/InjectAutoLoader/InjectAutoLoader.m:1]: Sure, I forgot about that :D I used to keep that header

**Comment by wojciech-kulik** [comment]
Could you also run some tests on your side before merging? To make sure it doesn't break anything.

File: Sources/InjectAutoLoader/InjectAutoLoader.m [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+@import Inject;
+
+@interface InjectAutoLoader: NSObject
++ (void)load;
+@end
+
+@implementation InjectAutoLoader
++ (void)load
+{
+#ifdef DEBUG
```
Comment (johnno1962) [inline Sources/InjectAutoLoader/InjectAutoLoader.m:12]: Can we put the #if DEBUG around everything in this file including the @import?

Comment (wojciech-kulik) [reply Sources/InjectAutoLoader/InjectAutoLoader.m]: I pushed updates. 

However, I'm not sure about the second part. I'm not very fluent with Objective-C, but I think it doesn't make a big difference, does it?

File: Sources/Inject/Inject.swift [hunk 5]
```diff
@@ -7,22 +7,19 @@ import SwiftUI
...
     associatedtype InjectInstanceType = Self
 
-    func enableInjection()
     func onInjection(callback: @escaping (InjectInstanceType) -> Void) -> Void
 }
 
 /// Public namespace for using Inject API
 public enum Inject {
     public static let observer = injectionObserver
-    public static let load: Void = loadInjectionImplementation
```
Comment (johnno1962) [inline Sources/Inject/Inject.swift:17]: Is it necessary to deprecate this released api?

Comment (wojciech-kulik) [reply Sources/Inject/Inject.swift:17]: Not necessary, but I wonder if we should keep it if it happens automatically?

Comment (wojciech-kulik) [reply Sources/Inject/Inject.swift:17]: I restored this method, but added "deprecated" warning instead for now :)

File: Sources/Inject/Inject.swift [hunk 6]
```diff
@@ -14,6 +14,9 @@ public protocol InjectListener {
 public enum Inject {
     public static let observer = injectionObserver
     public static var animation: SwiftUI.Animation?
+    
+    @available(*, deprecated, message: "You can now remove this call. It happens automatically when the app starts. This method will be removed in the future.")
+    public static let load: Void = loadInjectionImplementation
```
Comment (johnno1962) [inline Sources/Inject/Inject.swift:19]: Good compromise 👍

**Comment by wojciech-kulik** [comment]
@johnno1962 @krzysztofzablocki anything else to change or can we merge it? :)

**Comment by johnno1962** [comment]
I have mixed feelings about this change. Sometimes it is useful to be able to opt in and out of loading the injection bundle from the source code if you suspect you may be experiencing some strange artefact and would like to eliminate it as a possible cause. This is @krzysztofzablocki's project however and he'll be the one making the call.

**Comment by wojciech-kulik** [comment]
> I have mixed feelings about this change. Sometimes it is useful to be able to opt in and out of loading the injection bundle from the source code if you suspect you may be experiencing some strange artefact and would like to eliminate it as a possible cause. This is @krzysztofzablocki's project however and he'll be the one making the call.

Hmm, sounds reasonable. Maybe we could add some option to opt-out, but I'm not sure if it is possible with this approach.

**Comment by krzysztofzablocki** [comment]
yeah I'm not convinced this is needed tbh, this only saves you 1 line now right? and it takes away ability to opt in/out easily

**Comment by wojciech-kulik** [comment]
Ok, so I think we can drop it then :). 

In my case, I need it because I don't want to commit this line and I'm seeing changes in this file in git all the time. However, I've got this source code included locally anyway, so I can keep auto reload in my case.

**Pull request closed without merging**